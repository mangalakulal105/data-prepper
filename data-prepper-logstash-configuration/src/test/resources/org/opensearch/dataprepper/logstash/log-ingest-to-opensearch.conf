input {
    http {
        port => 3000
        threads => 10.0
        ssl => true
    }
}
filter {
    grok {
        match => {"log" => "%{COMBINEDAPACHELOG}"}
        break_on_match => false
    }
    drop { }
    kv {
        source => "message"
        target => "test"
    }
    mutate {
        add_field => { "message1" => 3 "message2" => 4.2 "message3" => "test3" "message4" => true }
        rename => { "messagea" => "newMessage" "messageb" => "newMessage2" }
        copy => { "messagec" => "messaged" }
        remove_field => ["test", "test2"]
        remove_field => ["test3", "test4"]
        uppercase => ["uppercaseField"]
        lowercase => ["lowercaseField"]
        trim => ["trimField"]
        gsub => ["source", "from", "to", "source2", "from2", "to2"]
        split => { "splitMessage" => ","}
    }
}
output {
    elasticsearch {
        hosts => ["https://localhost:19000"]
        user => myuser
        password => mypassword
        index => "simple-pipeline"
    }
}