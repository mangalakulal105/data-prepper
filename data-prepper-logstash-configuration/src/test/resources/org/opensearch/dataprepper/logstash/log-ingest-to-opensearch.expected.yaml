logstash-converted-pipeline:
  source:
    http:
      max_connection_count: 500
      request_timeout: 10000
      port: 3000
      thread_count: 10.0
      ssl: true
  processor:
    - grok:
        break_on_match: false
        match:
          log:
            - "%{COMBINEDAPACHELOG}"
    - drop_events: null
    - key_value:
        source: "message"
        destination: "test"
    - add_entries:
        entries:
          - key: "message1"
            value: 3
          - key: "message2"
            value: 4.2
          - key: "message3"
            value: "test3"
          - key: "message4"
            value: true
    - rename_keys:
        entries:
          - from_key: "messagea"
            to_key: "newMessage"
          - from_key: "messageb"
            to_key: "newMessage2"
    - copy_values:
        entries:
          - from_key: "messagec"
            to_key: "messaged"
    - delete_entries:
        with_keys:
          - "test"
          - "test2"
          - "test3"
          - "test4"
    - uppercase_string:
        with_keys:
          - "uppercaseField"
    - lowercase_string:
        with_keys:
          - "lowercaseField"
    - trim_string:
        with_keys:
          - "trimField"
    - substitute_string:
        entries:
          - source: "source"
            from: "from"
            to: "to"
          - source: "source2"
            from: "from2"
            to: "to2"
    - split_string:
        entries:
          - source: "splitMessage"
            delimiter: ","
  sink:
    - opensearch:
        hosts:
          - "https://localhost:19000"
        username: "myuser"
        password: "mypassword"
        index: "simple-pipeline"
